mlrun/serving/remote.py

```
@@ -527,3 +528,79 @@ def post_init(self, mode="sync", **kwargs):
         self.fill_placeholders = self.fill_placeholders or False
         self.rundb = mlrun.get_run_db()
         self.url = self.rundb.get_base_api_url(self.path)
+
+
+class RemoteFunctionStep(RemoteStep):
+    """
+    Graph step implementation for invoking functions remotely.
+    :param fn: Either an `mlrun.runtimes.RemoteRuntime` object or
+    a string URI in the form `function_name` or `project_name/function_name`.
+    :param project_name: Optional project name containing the function. If not provided,
+    the project name will be derived automatically according to the following order:
+    1. Extracted from the function URI (if specified as 'project_name/function_name')
+    2. Taken from the `project_name` parameter
+    3. Inferred from the current runtime or graph execution context
+    """
+
+    def __init__(
+        self,
+        fn: Union[mlrun.runtimes.RemoteRuntime, str, None] = None,
+        project_name: str = "",
+        **kwargs,
+    ):
+        super().__init__(url="", **kwargs)
+        self.rundb = None
+        self.fn = fn
+        self.project_name = project_name
+
+    def post_init(self, mode="sync", **kwargs):
+        self.rundb = mlrun.get_run_db()
+        if not isinstance(self.fn, (mlrun.runtimes.RemoteRuntime, str)):
+            raise mlrun.errors.MLRunInvalidArgumentTypeError(
+                "Parameter 'fn' must be of type mlrun.runtimes.RemoteRuntime or str."
+            )
+
+        if not self.fn:
+            raise mlrun.errors.MLRunRuntimeError(
+                "Parameter 'fn' can not be an empty string.\n"
+            )
+
+        if isinstance(self.fn, str):
+            project, uri, tag, hash_key = parse_versioned_object_uri(self.fn)
+
+            if self.project_name and project:
```

if `self.project_name == project` I don't think we should raise an error